### PR TITLE
type: fix notification type cannot jump

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | type: fix type cannot jump |
| 🇨🇳 Chinese | type: 修复鼠标点击 notification 方法不能跳转到 dts 文件的问题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/notification/index.tsx [hunk 1]
```diff
@@ -218,30 +218,34 @@ function destroy(key: React.Key) {
...
   _InternalPanelDoNotUseOrYouWillBeFired: typeof PurePanel;
-} = {
+}
+
+interface Methods {
+  success: (config: ArgsProps) => void;
+  info: (config: ArgsProps) => void;
+  warning: (config: ArgsProps) => void;
+  error: (config: ArgsProps) => void;
+}
```
Comment (MadCcc) [inline components/notification/index.tsx:235]:
```suggestion
interface Methods {
  [key: NoticeType]: (config: ArgsProps) => void;
}
```

File: components/notification/index.tsx [hunk 2]
```diff
@@ -218,30 +218,34 @@ function destroy(key: React.Key) {
...
   open: (config: ArgsProps) => void;
   destroy: (key?: React.Key) => void;
   config: any;
   useNotification: typeof useNotification;
   /** @private Internal Component. Do not use in your production. */
   _InternalPanelDoNotUseOrYouWillBeFired: typeof PurePanel;
-} = {
+}
+
+interface Methods {
```
Comment (MadCcc) [inline components/notification/index.tsx:230]:
```suggestion
interface NoticeMethods {
```

File: components/notification/index.tsx [hunk 3]
```diff
@@ -218,30 +218,34 @@ function destroy(key: React.Key) {
...
 
-methods.forEach((type) => {
-  staticMethods[type] = (config) =>
-    open({
-      ...config,
-      type,
-    });
+methods.forEach((type: MethodType) => {
+  staticMethods[type] = (config) => open({ ...config, type });
 });
```
Comment (MadCcc) [inline components/notification/index.tsx:249]: 可以试试 Object.assign

**Comment by MadCcc** [comment]
message 也是类似的实现

File: components/notification/index.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/notification/index.tsx:235]: 这样也没办法跳转

File: components/notification/index.tsx (hunk 3 above)
Comment (li-jia-nan) [reply components/notification/index.tsx:249]: 直接改成扩展运算符吧，看起来清晰一点，和 Object.assign 差不多

File: components/notification/index.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/notification/index.tsx:230]: done~

File: components/notification/index.tsx [hunk 4]
```diff
@@ -218,31 +216,37 @@ function destroy(key: React.Key) {
...
-      ...config,
-      type,
-    });
-});
+const staticMethods: BaseMethods & NoticeMethods = {
+  ...baseStaticMethods,
+  info: (config) => open({ ...config, type: 'info' }),
+  error: (config) => open({ ...config, type: 'error' }),
+  warning: (config) => open({ ...config, type: 'warning' }),
+  success: (config) => open({ ...config, type: 'success' }),
```
Comment (MadCcc) [inline components/notification/index.tsx:248]: 看起来不太聪明的写法😂

Comment (li-jia-nan) [reply components/notification/index.tsx:248]: 我改回 forEach 好了😂

Comment (li-jia-nan) [reply components/notification/index.tsx]: reduce 不太方便阅读，我改回一开始的 forEach 了，然后上面类型用 as 断言一下😂

**Comment by li-jia-nan** [comment]
> message 也是类似的实现

- #40864

**Review by zombieJ — APPROVED** [review]

**Pull request merged**